File: Assignment_39/program4.cpp
#include<iostream>
using namespace std;

template <class T>
int SearchLast(T *arr, int iSize, T iNo)
{
    int i = 0;

    for(i = iSize - 1; i > 0; i--)
    {
        if(arr[i] == iNo)
        {
            break;
        }
    }

    if(i == iSize)
    {
        return -1;
    }
    else
    {
        return i + 1;
    }
}

int main()
{
    int arr[] = {10,20,30,10,40,10,40,10};

    int iRet = SearchLast(arr,9,40);

    cout<<iRet;

    return 0;
}

File: Assignment_53/program4.java
import java.util.*;

class MyArray
{
    public int Arr[];

    public MyArray(int iSize)
    {
        Arr = new int[iSize];
    }

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);

        int iCnt = 0;

        System.out.println("Enter the array elements : ");
        for(iCnt = 0; iCnt < this.Arr.length; iCnt++)
        {
            this.Arr[iCnt] = sobj.nextInt();
        }
    }

    public void Display()
    {
        int iCnt = 0;

        System.out.println("Elements of the array are : ");
        for(iCnt = 0; iCnt < this.Arr.length; iCnt++)
        {
            System.out.print(this.Arr[iCnt]+"\t");
        }
        System.out.println();
    }

    public int[] CopyArray()
    {
        int iCnt = 0;
        int NewArray[] = new int[Arr.length];

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            NewArray[iCnt] = Arr[iCnt];
        }

        return NewArray;
    }
}

class program4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the number of elements : ");
        int iSize = sobj.nextInt();

        MyArray mobj = new MyArray(iSize);

        mobj.Accept();

        mobj.Display();

        int[] iRet = mobj.CopyArray();
        System.out.println("Copied Array : "+Arrays.toString(iRet));
    }
}